reader/src/main/java/org/jline/reader/impl/LineReaderImpl.java

```diff
@@ -586,7 +590,11 @@ public String readLine(String prompt, String rightPrompt, MaskingCallback maskin
 
             smallTerminalOffset = 0;
 
-            state = State.NORMAL;
+            if (isDumb && "\n".equals(getLastBinding())) {
```

Comment:
Could you add 2 constants for `CR = "\r"` and `LF = "\n"`, that would make the code more readable when talking about _ignoring CR_.

Reply:
Added along with some more comments